R/methods-Chromatograms.R

```diff
@@ -190,3 +212,52 @@ setMethod("plot", signature = signature("Chromatograms"),
                            xlab = xlab, ylab = ylab, main = main, ...)
               }
           })
+
+#' @rdname Chromatograms-class
+#'
+#' @description \code{phenoData}: accesses the phenotypical desccription of the
+#'     samples. Returns an \code{NAnnotatedDataFrame} object.
+setMethod("phenoData", "Chromatograms", function(object) object@phenoData)
+
+#' @rdname Chromatograms-class
+#'
+#' @description \code{pData}: accesses the phenotypical description of the
+#'     samples. Returns a \code{data.frame}.
+setMethod("pData", "Chromatograms", function(object) pData(phenoData(object)))
+
+#' @rdname Chromatograms-class
+#'
+#' @description \code{pData<-}: replace the phenotype data.
+setReplaceMethod("pData", "Chromatograms", function(object, value) {
+    pData(object@phenoData) <- value
+    if (any(rownames(value) != as.character(1:nrow(value))))
+        colnames(object@.Data) <- rownames(value)
```

Comment:
Now I realise that there is this issue in `MSnSet`s too
```r
> data(msnset)
> sampleNames(msnset)
[1] "iTRAQ4.114" "iTRAQ4.115" "iTRAQ4.116" "iTRAQ4.117"
> pData(msnset)
              mz reporters
iTRAQ4.114 114.1    iTRAQ4
iTRAQ4.115 115.1    iTRAQ4
iTRAQ4.116 116.1    iTRAQ4
iTRAQ4.117 117.1    iTRAQ4
> pd <- data.frame(x = 1:4, y = 4:1)
> rownames(pd) <- LETTERS[1:4]
> pd
  x y
A 1 4
B 2 3
C 3 2
D 4 1
> pData(msnset) <- pd
> sampleNames(msnset)
[1] "A" "B" "C" "D"
> validObject(msnset)
Error in validObject(msnset) : 
  invalid class “MSnSet” object: 1: sampleNames differ between assayData and phenoData
invalid class “MSnSet” object: 2: sampleNames differ between phenoData and protocolData
```
I think that the error should be thrown when trying to assign a new `pData` with different rownames.